### PR TITLE
Expect comparison

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #884 

Based against #931 since it uses the `infix_metadata` data.frame from there.

### CODE REVIEW

File: R/expect_comparison_linter.R [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+expect_comparison_linter <- function() {
+  Linter(function(source_file) {
+    if (length(source_file$parsed_content) == 0L) {
+      return(list())
+    }
+
+    xml <- source_file$xml_parsed_content
+
+    # != doesn't have a clean replacement
+    comparator_nodes <-
```
Comment (AshesITR) [inline R/expect_comparison_linter.R:21]: unused lint due to `glue`.

File: R/expect_comparison_linter.R [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      setdiff(as.list(infix_metadata$xml_tag[infix_metadata$comparator]), "NE")  # nolint: object_usage_linter.
+    xpath <- glue::glue("//expr[
+      expr[SYMBOL_FUNCTION_CALL[text() = 'expect_true']]
+      and expr[2][ {do.call(xp_or, comparator_nodes)} ]
```
Comment (AshesITR) [inline R/expect_comparison_linter.R:25]: better extend `xp_or()` to be vectorized?

Comment (MichaelChirico) [reply R/expect_comparison_linter.R:25]: I'll write that as a separate PR. I had started to do that for another PR but it did look messy. Probably worth it though.

Comment (MichaelChirico) [reply R/expect_comparison_linter.R:25]: #961

File: R/expect_comparison_linter.R (hunk 1 above)
Comment (MichaelChirico) [reply R/expect_comparison_linter.R:21]: ah I have the `# nolint` on the next line. moved.

File: R/expect_comparison_linter.R [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  `<` = "expect_lt", `<=` = "expect_lte",
+  `==` = "expect_identical"
+)
+
+gen_expect_comparison_lint <- function(expr, source_file) {
+  comparator <- xml2::xml_text(xml2::xml_find_first(expr, "expr[2]/*[2]"))
+  expectation <- comparator_expectation_map[[comparator]]
+  lint_msg <- sprintf("%s(x, y) is better than expect_true(x %s y).", expectation, comparator)
+  xml_nodes_to_lint(expr, source_file, lint_msg, type = "warning")
+}
```
Comment (AshesITR) [inline R/expect_comparison_linter.R:44]: This can use the new `xml_nodes_to_lint` functionality IINM

**Review by AshesITR — APPROVED** [review]

**Pull request merged**